### PR TITLE
Add test for TensorRT version-compatible model support

### PR DESCRIPTION
**Pull request by dyastremsky**
Add testing for TensorRT version-compatible model support.

The feature will be implemented in the TensorRT backend in this pull request: https://github.com/triton-inference-server/tensorrt_backend/pull/74.

### CODE REVIEW

File: qa/L0_trt_compat/test.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+SERVER_ARGS="--model-repository=`pwd`/models --exit-timeout-secs=120 --backend-config=tensorrt,version-compatible=true"
+
+run_server
+if [ "$SERVER_PID" == "0" ]; then
+    cat $SERVER_LOG
+    echo -e "\n***\n*** FAILED: unsuccessful server start (version compatibility enabled): $SERVER\n***"
+    exit 1
+fi
+
```
Comment (tanmayv25) [inline qa/L0_trt_compat/test.sh:70]: Should kill and wait for the server in success case.

Comment (dyastremsky) [reply qa/L0_trt_compat/test.sh:75]: Thanks, added.

[the remark this answers is not in the document]
Comment (dyastremsky) [reply qa/L0_trt_compat/test.sh:75]: Sure, added!

File: qa/L0_trt_compat/test.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    REPO_VERSION=$1
+fi
+if [ -z "$REPO_VERSION" ]; then
+    echo -e "Repository version must be specified"
+    echo -e "\n***\n*** Test Failed\n***"
+    exit 1
+fi
+
+TEST_RESULT_FILE='test_results.txt'
+COMPATABILITY_TEST_PY=trt_compatibility_test.py
```
Comment (rmccorm4) [inline qa/L0_trt_compat/test.sh:39]: nit: COMPATIBILITY spelling everywhere

File: qa/L0_trt_compat/test.sh [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+run_server
+if [ "$SERVER_PID" == "0" ]; then
+    cat $SERVER_LOG
+    echo -e "\n***\n*** FAILED: unsuccessful server start (version compatibility enabled): $SERVER\n***"
+    exit 1
+fi
+
+set +e
+
+python $COMPATABILITY_TEST_PY TrtCompatabilityTest >$CLIENT_LOG 2>&1
```
Comment (rmccorm4) [inline qa/L0_trt_compat/test.sh:78]: spelling

File: qa/L0_trt_compat/test.sh [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+fi
+
+run_server
+if [ "$SERVER_PID" != "0" ]; then
+    cat $SERVER_LOG
+    echo -e "\n***\n*** FAILED: unexpected server start (version compatibility disabled): $SERVER\n***" >> $CLIENT_LOG
+    kill $SERVER_PID
+    wait $SERVER_PID
+    exit 1
+fi
```
Comment (rmccorm4) [inline qa/L0_trt_compat/test.sh:65]: Can you grep server log for expected message? We have a lot of common TRT related model errors/failures in CI, so this part could pass just for model not being found or something.

Comment (dyastremsky) [reply qa/L0_trt_compat/test.sh:65]: Great suggestion! Done.

File: qa/L0_trt_compat/test.sh (hunk 3 above)
Comment (dyastremsky) [reply qa/L0_trt_compat/test.sh]: Nice catch, fixed!

File: qa/L0_trt_compat/test.sh (hunk 2 above)
Comment (dyastremsky) [reply qa/L0_trt_compat/test.sh]: Fixed.

**Review by rmccorm4 — APPROVED** [review]

**Pull request merged**